codegen/model/types/parser.go

```diff
@@ -263,6 +274,15 @@ func (p *Parser) getFieldType(goType reflect.Type, protoField protoreflect.Field
 	}
 
 	if mirTypeOption != "" {
+		sepMapIdx := strings.Index(mirTypeOption, "|")
+		if sepMapIdx != -1 { // it is a map type
+			if !mapType { //key value
+				mirTypeOption = mirTypeOption[:sepMapIdx]
+			} else { // map value
+				mirTypeOption = mirTypeOption[sepMapIdx+1:]
+			}
+		}
```

Comment:
Also, I would be curious to know your opinion on potential alternative syntaxes for the annotation.

A few natural examples (in no specific order):
1. using -> as the separator and trimming the whitespaces around it
2. using map<keyType, valueType> to follow the syntax of protobufs
3. using map[keyType]valueType to follow the syntax of Go
4. using two separate annotations (mir.key_type) and (mir.value_type) and throwing an error if (mir.type) is used for a map type

One potential concern about all options except the last one (including the current version) is the length of the annotation, since, as of now, the type annotation has to include the full package path for each type.

Reply:
Option (4) could look somewhat like this:

```protobuf
message MapMessage {
    map<string, int> mapValue [
        (mir.key_type) = "github.com/awesome-company/awesome-project/package.KeyType",
        (mir.value_type) = "github.com/awesome-company/awesome-project/package.ValueType",
    ];
}
```

The current version may be a bit hard to parse visually:

```protobuf
message MapMessage {
    map<string, int> mapValue [(mir.type) = "github.com/awesome-company/awesome-project/package.KeyType|github.com/awesome-company/awesome-project/package.ValueType"];
}
```

It could probably be helped to a certain extent by trimming the whitespaces around the separator.

Reply:
Intuitively, option (4) seems the nicest to me. On top of what you already said, it seems to avoid the necessity of string parsing magic. @ranchalp what do you think?

Reply:
Option 4 sounds good. Just did it, and took the extra effort to also support defining only the key or only the value types (which is related to other comments of this PR and will enable for removing even more boilerplate code).

Reply:
I also left as TODO in issue #356 the last bit that I think we need to completely convert all proto messages with the codegen. Will be doing that in an upcoming PR. Thanks for your comments! :)

Reply:
Sorry, I just realized that there is actually a perfectly valid use-case for using all three of `(mir.value_type)`, `(mir.key_type)`, and `(mir.type)`. Namely, when there is a wrapped map type.

Something like:
```go
// Membership represents the list of participants with their respective weights.
type Membership map[NodeID]int
```

Reply:
ah, but maybe it's already supported by the current code :)

Reply:
No, I guess it's not, but it should be the matter of removing the extra check

Reply:
Hm, there is an issue though: it won't be consistent with how `(mir.type)` operates on slices

Reply:
maybe, in the long term (perhaps, not in this PR), it would be better to migrate slices to using a pair of annotations:
1. `(mir.value_type)` for the underlying type
2. `(mir.type)` for wrapping the slice itself

Reply:
Could that be somehow related to a solution to #356 ?

But I agree that this should definitely be addressed in a different PR.